app/src/main/java/tech/zafrani/companionforpubg/fragments/NewsFragment.java

```diff
@@ -46,14 +44,17 @@ public void onViewCreated(@NonNull final View view,
         adapter =  new NewsAdapter();
         recyclerView.setAdapter(adapter);
 
-        final PUBGNewsFetch newsFetch = new PUBGNewsFetch();
-        newsFetch.delegate = this;
+        final PUBGNewsFetch newsFetch = new PUBGNewsFetch(this);
         newsFetch.execute();
     }
 
     @Override
     public void onDestroy() {
         super.onDestroy();
+        if (recyclerView != null) {
+            recyclerView.setAdapter(null);
```

Comment:
this needs to be done before super.onDestroy().

super.onDestroy will call the BaseFragments onDestroy method which uses butterknifes unbind method. The unbind method sets anything binded with butterknife to null. So after super.ondestroy recyclerview will be null but the instance of it in memory will still hold a reference to the adapter.